app/bundles/CoreBundle/Service/BatchDeleteService.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Mautic\CoreBundle\Service;
+
+use Mautic\CoreBundle\Model\FormModel;
+use Mautic\CoreBundle\Model\MauticModelInterface;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Translation\Translator;
+use Symfony\Component\HttpFoundation\Request;
+
+final class BatchDeleteService
+{
+    public function __construct(
+        private readonly CorePermissions $security,
+        private readonly Translator $translator,
+    ) {
+    }
+
+    /**
+     * Perform batch delete on entities.
+     *
+     * @param FormModel $model
+     *
+     * @return array
+     *               Array of flash messages
+     */
+    public function batchDelete(
+        Request $request,
+        MauticModelInterface $model,
+        array $postActionVars,
+        string $sessionBase,
+        string $modelName,
+        callable $isLocked,
+    ): array {
+        $ids            = $request->query->get('ids');
```

Comment:
Instead of sending the request object, send the `ids` and the `search` as params here. You'll be able to remove the `$sessionBase` too this way.